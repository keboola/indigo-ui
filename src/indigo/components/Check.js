import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/free-solid-svg-icons/faCheck';
import { faTimes } from '@fortawesome/free-solid-svg-icons/faTimes';

class Check extends React.Component {
  render() {
    return (
      <FontAwesomeIcon
        icon={this.props.isChecked ? faCheck : faTimes}
        fixedWidth={this.props.fixedWidth}
        className={classnames(this.props.className)}
      />
    );
  }
}

Check.propTypes = {
  isChecked: PropTypes.bool.isRequired,
  fixedWidth: PropTypes.bool,
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

Check.defaultProps = {
  fixedWidth: false,
};

export default Check;
