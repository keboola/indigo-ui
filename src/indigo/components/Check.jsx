import React from 'react';
import PropTypes from 'prop-types';

const Check = React.createClass({
  propTypes: {
    isChecked: PropTypes.bool.isRequired
  },

  render() {
    return (
      <i className={this.props.isChecked ? 'fa fa-fw fa-check' : 'fa fa-fw fa-times'}/>
    );
  }
});

export default Check;
