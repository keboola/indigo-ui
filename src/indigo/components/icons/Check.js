import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Check = ({ className, ...props }) => {
  return (
    <svg
      width="16"
      height="12"
      viewBox="0 0 16 12"
      className={classNames('icon', className)}
      {...props}
    >
      <g fill="none" fillRule="evenodd">
        <path d="M-4-6h24v24H-4z" />
        <path
          d="M15.71.946a1.02 1.02 0 0 0-1.402 0l-8.41 8.086L1.691 4.99a1.02 1.02 0 0 0-1.402 0 .927.927 0 0 0 0 1.348l4.906 4.717a1.02 1.02 0 0 0 1.402 0l9.112-8.76a.927.927 0 0 0 0-1.348z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </g>
    </svg>
  );
};

Check.propTypes = {
  className: PropTypes.string
};

export default Check;
