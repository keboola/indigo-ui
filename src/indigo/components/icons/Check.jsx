import React from 'react';

const Check = props => (
  <svg width={16} height={12} viewBox="0 0 16 12" {...props}>
    <title>ic_check</title>
    <g fill="none" fillRule="evenodd">
      <path d="M-4-6h24v24H-4z" />
      <path
        d="M15.71.946a1.02 1.02 0 0 0-1.402 0l-8.41 8.086L1.691 4.99a1.02 1.02 0 0 0-1.402 0 .927.927 0 0 0 0 1.348l4.906 4.717a1.02 1.02 0 0 0 1.402 0l9.112-8.76a.927.927 0 0 0 0-1.348z"
        fill="currentColor"
        fillRule="nonzero"
      />
    </g>
  </svg>
);

export default Check;
