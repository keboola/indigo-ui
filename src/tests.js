import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
