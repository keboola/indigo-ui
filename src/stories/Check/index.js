import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';

import Check from '../../indigo/components/Check';

storiesOf('Check', module).add(
  'Check',
  () => {
    const bsStyle = boolean('isChecked', true);
    return <Check isChecked={bsStyle} />;
  },
  { info: { text: 'Show if something is enabled or not.' } }
);
