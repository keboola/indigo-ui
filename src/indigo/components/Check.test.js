import React from 'react';

import { snapshot } from "../../tests";
import Check from './Check';

describe('<Check />', () => {
  it('Checkbox - checked', () => {
    snapshot(<Check isChecked />);
  });

  it('Checkbox - not checked', () => {
    snapshot(<Check isChecked={false} />);
  });
});
