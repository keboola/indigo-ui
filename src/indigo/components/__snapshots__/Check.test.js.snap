// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Check /> Checkbox - checked 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-check fa-w-16 "
  data-icon="check"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 512 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;

exports[`<Check /> Checkbox - not checked 1`] = `
<svg
  aria-hidden="true"
  className="svg-inline--fa fa-times fa-w-11 "
  data-icon="times"
  data-prefix="fas"
  focusable="false"
  role="img"
  style={Object {}}
  viewBox="0 0 352 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
    fill="currentColor"
    style={Object {}}
  />
</svg>
`;
